dwm: cleanup

06kellyjac opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Dropped `dwm-git` since it hasn't been updated for years even thought it's meant to be trailing master
If people want `dwm-git` (or `dmenu-git` & `st-git`) I've made a neovim-nightly style overlay https://github.com/06kellyjac/suckless-git-overlay
LMK if the throw message needs changing

Cleaned up `dwm`

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS (x86_64)
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

06kellyjac committed:
dwm-git: drop

06kellyjac commented:
Result of `nixpkgs-review pr 112299` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dwm</li>
  </ul>
</details>

AndersonTorres commented:
If the git version is not being updated in 2 years, then it looks more logical to me it is the true stable one.

06kellyjac commented:
I haven't seen any stability concerns over 6.2 and it's not exactly a new release coming out 2019-02-02, and there are a lot of projects based off dwm at some point or another.
If it is a stable version it should be called dwm-stable-DATE IMO

@xeji would probably remember but they've not been active on GH for a while

Mic92 commented:
> I haven't seen any stability concerns over 6.2 and it's not exactly a new release coming out 2019-02-02, and there are a lot of projects based off dwm at some point or another.
> If it is a stable version it should be called dwm-stable-DATE IMO
> 
> @xeji would probably remember but they've not been active on GH for a while

I think this was meant as a joke.

AndersonTorres commented:
@06kellyjac ping

ivankovnatsky commented:
Hi Jack, I switched to i3, not interested in maintaining dwm* packages as of now. Thanks.

06kellyjac commented:
No worries :+1:

06kellyjac commented on pkgs/top-level/all-packages.nix:
```diff
@@ -21670,11 +21670,6 @@ in
     patches = config.dwm.patches or [];
```
Is this actually missing `conf = config.dwm.conf or null;` that `dwm-git` has?
How does `conf ? null` get populated otherwise...

Should that be added in this PR too or another PR?

@AndersonTorres

AndersonTorres replied:
How many packages still use the config interface? I think it is 'reserved' for NixOS or external programs like Home Manager.

User-provided patches are usually a parameter with null as default.

06kellyjac replied:
:shrug: I have no idea

I just saw `conf` was on the `dwm-git` copy but not on `dwm` but they both include it on the first line.

AndersonTorres replied:
Get rid of that config reference.

06kellyjac replied:
The patches = config.dwm.patches part?
Do i also need to change the nixos module?

AndersonTorres replied:
The `patches = config.dwm.patches` reference.

Is the module change needed?

06kellyjac replied:
I don't know, thats why I'm asking :sweat_smile:
I dont have much experience with nixos modules

Mic92 replied:
I think we can keep it here for backwards compatibility. Long-term we hopefully get a better nix interface for expressing packages overrides.

AndersonTorres replied:
I have read the module, and it has no reference about configuring patches. Then it needs no modification at all.

06kellyjac replied:
Ok I've removed it. LMK if that's right

Mic92 replied:
It might break people still that used this option before.

06kellyjac replied:
I guess if someone runs into the issue we add it back in?
Or do we direct them to do it a different way such as with overrides?

neonfuz replied:
I don't understand why the config reference was removed, I use nixpkgs.config.dwm.patches to set my patches and this needlessly broke my config on 21.05 upgrade. I'm making an issue to add this feature back.

06kellyjac committed:
dwm: cleanup

- nixpkgs-fmt
- use pname and version
- use multiline for prePatch to make tree-wide modifications easier
- remove global `with lib;`
- update homepage to dwm page
- copy description from git repo
- add longDescription
- remove unused config.dwm.patches
